### PR TITLE
Close the client pipe on shutdown

### PR DESCRIPTION
**Pull request by sroache**
When shutting down the extension the named pipe that is created by the extension is closed however the "client" (the handle to the pipe created by osquery - used for pings) is not closed. This leads to osquery not closing correctly when the extension is still connected.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -258,6 +263,16 @@ func (s *ExtensionManagerServer) Shutdown(ctx context.Context) error {
 		}()
 	}
 
+	// Also close the client that is attached to the osquery hosted pipe
+	if s.serverClient != nil {
+		serverClient := s.serverClient
+		s.serverClient = nil
+
+		go func() {
```
Comment (zwass) [inline server.go]: Can you explain why this happens in a separate goroutine rather than synchronously in this function?

Comment (sroache) [reply server.go]: I just tested it without the routine and it worked fine - removed the routine now.
I originally just copied the flow as was happening with the server above.

**Comment by sroache** [comment]
@zwass Sorry this took so long. The CLA has been signed now. Is there a way to re-trigger the build?

**Comment by zwass** [comment]
@sroache you can push new changes to the branch to trigger a rebuild, or we will have to ask @directionless to trigger it (I no longer have write access to this repository).

**Comment by sroache** [comment]
@directionless Both this PR and develop appear to be failing for the same reason. Do you know anything about this?

**Review by directionless — COMMENTED** [review]
I think the intent here is reasonable, but I think the close behavior needs to take into account whether the client was created in-sito vs passed in via #107

**Review by directionless — COMMENTED** [review]
I think the intent here is reasonable, but I think the close behavior needs to take into account whether the client was created in-sito vs passed in via #107

**Pull request closed without merging**